From 80077209745285f0e1549cd0f6fe542c65ce86e8 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Wed, 2 Dec 2020 03:40:38 +0100
Subject: [PATCH] iwinfo: add a few more Wi-Fi cards like Atheros, Marvell,
 Mikrotik

This will help to users mostly in LuCI that these cards will no longer
be shown as "Generic"

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 package/network/utils/iwinfo/Makefile         |  2 +-
 ...o-add-device-id-for-MediaTek-MT7615E.patch | 26 ++++++++++++++
 ...nfo-add-device-id-for-Atheros-AR9287.patch | 26 ++++++++++++++
 ...vice-id-for-Qualcomm-Atheros-QCA9990.patch | 28 +++++++++++++++
 ...e-id-for-Mikrotik-R11e-5HacD-miniPCI.patch | 34 +++++++++++++++++++
 ...e-id-for-Atheros-AR9287-PCIe-wifi-ca.patch | 34 +++++++++++++++++++
 ...e-id-for-Marvell-88W8997-SDIO-wifi-c.patch | 33 ++++++++++++++++++
 7 files changed, 182 insertions(+), 1 deletion(-)
 create mode 100644 package/network/utils/iwinfo/patches/0001-iwinfo-add-device-id-for-MediaTek-MT7615E.patch
 create mode 100644 package/network/utils/iwinfo/patches/0002-iwinfo-add-device-id-for-Atheros-AR9287.patch
 create mode 100644 package/network/utils/iwinfo/patches/0003-iwinfo-add-device-id-for-Qualcomm-Atheros-QCA9990.patch
 create mode 100644 package/network/utils/iwinfo/patches/0004-iwinfo-add-device-id-for-Mikrotik-R11e-5HacD-miniPCI.patch
 create mode 100644 package/network/utils/iwinfo/patches/0005-iwinfo-add-device-id-for-Atheros-AR9287-PCIe-wifi-ca.patch
 create mode 100644 package/network/utils/iwinfo/patches/0006-iwinfo-add-device-id-for-Marvell-88W8997-SDIO-wifi-c.patch

diff --git a/package/network/utils/iwinfo/Makefile b/package/network/utils/iwinfo/Makefile
index cee7253a3a..bd4bac7f08 100644
--- a/package/network/utils/iwinfo/Makefile
+++ b/package/network/utils/iwinfo/Makefile
@@ -7,7 +7,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libiwinfo
-PKG_RELEASE:=1
+PKG_RELEASE:=3
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/iwinfo.git
diff --git a/package/network/utils/iwinfo/patches/0001-iwinfo-add-device-id-for-MediaTek-MT7615E.patch b/package/network/utils/iwinfo/patches/0001-iwinfo-add-device-id-for-MediaTek-MT7615E.patch
new file mode 100644
index 0000000000..d786d62a86
--- /dev/null
+++ b/package/network/utils/iwinfo/patches/0001-iwinfo-add-device-id-for-MediaTek-MT7615E.patch
@@ -0,0 +1,26 @@
+From 797b06916bba03822efd16c53764be669cad2d93 Mon Sep 17 00:00:00 2001
+From: Pawel Dembicki <paweldembicki@gmail.com>
+Date: Tue, 28 Jan 2020 22:16:43 +0100
+Subject: [PATCH 1/6] iwinfo: add device id for MediaTek MT7615E
+
+Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>
+(cherry picked from commit 7483398929d14e97a0b878af7e2432ae00f61c20)
+---
+ hardware.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/hardware.txt b/hardware.txt
+index e85db06..c5272b9 100644
+--- a/hardware.txt
++++ b/hardware.txt
+@@ -171,6 +171,7 @@
+ 0x14c3 0x7603 0x14c3 0x7603    0      0  "MediaTek" "MT7603E"
+ 0x14c3 0x7610 0x14c3 0x7610    0      0  "MediaTek" "MT7610E"
+ 0x14c3 0x7612 0x14c3 0x7612    0      0  "MediaTek" "MT7612E"
++0x14c3 0x7615 0x7615 0x14c3    0      0  "MediaTek" "MT7615E"
+ 0x14c3 0x7620 0x14c3 0x000c    0      0  "MediaTek" "MT7620"
+ 0x14c3 0x7628 0x14c3 0x0004    0      0  "MediaTek" "MT76x8"
+ 0x14c3 0x7650 0x14c3 0x7650    0      0  "MediaTek" "MT7610E"
+-- 
+2.25.1
+
diff --git a/package/network/utils/iwinfo/patches/0002-iwinfo-add-device-id-for-Atheros-AR9287.patch b/package/network/utils/iwinfo/patches/0002-iwinfo-add-device-id-for-Atheros-AR9287.patch
new file mode 100644
index 0000000000..68723f1593
--- /dev/null
+++ b/package/network/utils/iwinfo/patches/0002-iwinfo-add-device-id-for-Atheros-AR9287.patch
@@ -0,0 +1,26 @@
+From 36f5a35e9f2b6648d96369dc9c2c05c1c9689eac Mon Sep 17 00:00:00 2001
+From: Pawel Dembicki <paweldembicki@gmail.com>
+Date: Tue, 28 Jan 2020 22:16:44 +0100
+Subject: [PATCH 2/6] iwinfo: add device id for Atheros AR9287
+
+Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>
+(cherry picked from commit bb216982951698833bbdf4a88872e9b5ccd026a5)
+---
+ hardware.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/hardware.txt b/hardware.txt
+index c5272b9..42aa335 100644
+--- a/hardware.txt
++++ b/hardware.txt
+@@ -145,6 +145,7 @@
+ 0x168c 0x002a 0x168c 0xa093    0      0  "Atheros"  "AR9280"
+ 0x168c 0x002b 0x168c 0xa091    0      0  "Atheros"  "AR9285"
+ 0x168c 0x002d 0x168c 0x209a    0      0  "Atheros"  "AR9287"
++0x168c 0x002e 0x1a3b 0x1121    0      0  "Atheros"  "AR9287"
+ 0x168c 0x002e 0x0777 0xe0a2    8      0  "Ubiquiti" "NanoStation Loco M2 (XM)" /* wrong offset! */
+ 0x168c 0x0030 0x168c 0x3114    0      0  "Atheros"  "AR9390"
+ 0x168c 0x0033 0x168c 0xa120    0      0  "Atheros"  "AR9580"
+-- 
+2.25.1
+
diff --git a/package/network/utils/iwinfo/patches/0003-iwinfo-add-device-id-for-Qualcomm-Atheros-QCA9990.patch b/package/network/utils/iwinfo/patches/0003-iwinfo-add-device-id-for-Qualcomm-Atheros-QCA9990.patch
new file mode 100644
index 0000000000..5d3f353276
--- /dev/null
+++ b/package/network/utils/iwinfo/patches/0003-iwinfo-add-device-id-for-Qualcomm-Atheros-QCA9990.patch
@@ -0,0 +1,28 @@
+From 259a4fb7f23ab68578012dc1486bc3cbfe12702b Mon Sep 17 00:00:00 2001
+From: Robert Marko <robert.marko@sartura.hr>
+Date: Thu, 30 Jan 2020 22:47:29 +0100
+Subject: [PATCH 3/6] iwinfo: add device id for Qualcomm Atheros QCA9990
+
+Lets add the PCI ID for QCA9990
+
+Signed-off-by: Robert Marko <robert.marko@sartura.hr>
+(cherry picked from commit 9a4bae898f770fdf00858ef468d26a94367515f2)
+---
+ hardware.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/hardware.txt b/hardware.txt
+index 42aa335..1411206 100644
+--- a/hardware.txt
++++ b/hardware.txt
+@@ -155,6 +155,7 @@
+ 0x168c 0x0033 0x19b6 0xd016    0      0  "MikroTik" "R11e-2HPnD"
+ 0x168c 0x003c 0x0000 0x0000    0      0  "Qualcomm Atheros" "QCA9880"
+ 0x168c 0x003c 0x19b6 0xd03c    0      0  "Mikrotik" "R11e-5HacT"
++0x168c 0x0040 0x168c 0x0002    0      0  "Qualcomm Atheros" "QCA9990"
+ 0x168c 0x0046 0x168c 0xcafe    0      0  "Qualcomm Atheros" "QCA9984"
+ 0x168c 0x0050 0x0000 0x0000    0      0  "Qualcomm Atheros" "QCA9887"
+ 0x168c 0x0056 0x0000 0x0000    0      0  "Qualcomm Atheros" "QCA9886"
+-- 
+2.25.1
+
diff --git a/package/network/utils/iwinfo/patches/0004-iwinfo-add-device-id-for-Mikrotik-R11e-5HacD-miniPCI.patch b/package/network/utils/iwinfo/patches/0004-iwinfo-add-device-id-for-Mikrotik-R11e-5HacD-miniPCI.patch
new file mode 100644
index 0000000000..d15428fc77
--- /dev/null
+++ b/package/network/utils/iwinfo/patches/0004-iwinfo-add-device-id-for-Mikrotik-R11e-5HacD-miniPCI.patch
@@ -0,0 +1,34 @@
+From 6256efe0240cc98a74a92363b5528798e3694ecf Mon Sep 17 00:00:00 2001
+From: Johann Neuhauser <johann@it-neuhauser.de>
+Date: Mon, 6 Apr 2020 23:58:43 +0200
+Subject: [PATCH 4/6] iwinfo: add device id for Mikrotik R11e-5HacD miniPCIe
+ card
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Lets add device id for Mikrotik R11e-5HacD miniPCIe card.
+
+Signed-off-by: Johann Neuhauser <johann@it-neuhauser.de>
+[added missing commit description]
+Signed-off-by: Petr Štetiar <ynezz@true.cz>
+(cherry picked from commit 2faa20e5e9d107b97e393a4eb458370e80b4d720)
+---
+ hardware.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/hardware.txt b/hardware.txt
+index 1411206..ea03232 100644
+--- a/hardware.txt
++++ b/hardware.txt
+@@ -155,6 +155,7 @@
+ 0x168c 0x0033 0x19b6 0xd016    0      0  "MikroTik" "R11e-2HPnD"
+ 0x168c 0x003c 0x0000 0x0000    0      0  "Qualcomm Atheros" "QCA9880"
+ 0x168c 0x003c 0x19b6 0xd03c    0      0  "Mikrotik" "R11e-5HacT"
++0x168c 0x003c 0x19b6 0xd075    0      0  "Mikrotik" "R11e-5HacD"
+ 0x168c 0x0040 0x168c 0x0002    0      0  "Qualcomm Atheros" "QCA9990"
+ 0x168c 0x0046 0x168c 0xcafe    0      0  "Qualcomm Atheros" "QCA9984"
+ 0x168c 0x0050 0x0000 0x0000    0      0  "Qualcomm Atheros" "QCA9887"
+-- 
+2.25.1
+
diff --git a/package/network/utils/iwinfo/patches/0005-iwinfo-add-device-id-for-Atheros-AR9287-PCIe-wifi-ca.patch b/package/network/utils/iwinfo/patches/0005-iwinfo-add-device-id-for-Atheros-AR9287-PCIe-wifi-ca.patch
new file mode 100644
index 0000000000..9a3b3d9cc8
--- /dev/null
+++ b/package/network/utils/iwinfo/patches/0005-iwinfo-add-device-id-for-Atheros-AR9287-PCIe-wifi-ca.patch
@@ -0,0 +1,34 @@
+From aa85b8833ecfa39ac3d8378f424bf7dd51985420 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Pali=20Roh=C3=A1r?= <pali@kernel.org>
+Date: Wed, 20 May 2020 14:31:46 +0200
+Subject: [PATCH 5/6] iwinfo: add device id for Atheros AR9287 PCIe wifi card
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+This card is identified by lspci as:
+
+  01:00.0 Network controller [0280]: Qualcomm Atheros AR9287 Wireless Network Adapter (PCI-Express) [168c:002e] (rev 01)
+          Subsystem: Qualcomm Atheros Device [168c:30a4]
+
+Signed-off-by: Pali Rohár <pali@kernel.org>
+(cherry picked from commit f6b7d16d2ffa5e9fa235af4a79647637a866f635)
+---
+ hardware.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/hardware.txt b/hardware.txt
+index ea03232..d11d762 100644
+--- a/hardware.txt
++++ b/hardware.txt
+@@ -147,6 +147,7 @@
+ 0x168c 0x002d 0x168c 0x209a    0      0  "Atheros"  "AR9287"
+ 0x168c 0x002e 0x1a3b 0x1121    0      0  "Atheros"  "AR9287"
+ 0x168c 0x002e 0x0777 0xe0a2    8      0  "Ubiquiti" "NanoStation Loco M2 (XM)" /* wrong offset! */
++0x168c 0x002e 0x168c 0x30a4    0      0  "Atheros"  "AR9287"
+ 0x168c 0x0030 0x168c 0x3114    0      0  "Atheros"  "AR9390"
+ 0x168c 0x0033 0x168c 0xa120    0      0  "Atheros"  "AR9580"
+ 0x168c 0x0033 0x168c 0xa136    0      0  "Atheros"  "AR9580"
+-- 
+2.25.1
+
diff --git a/package/network/utils/iwinfo/patches/0006-iwinfo-add-device-id-for-Marvell-88W8997-SDIO-wifi-c.patch b/package/network/utils/iwinfo/patches/0006-iwinfo-add-device-id-for-Marvell-88W8997-SDIO-wifi-c.patch
new file mode 100644
index 0000000000..83037c8187
--- /dev/null
+++ b/package/network/utils/iwinfo/patches/0006-iwinfo-add-device-id-for-Marvell-88W8997-SDIO-wifi-c.patch
@@ -0,0 +1,33 @@
+From e3ec0af21cd791707fbdebd06f5ec0dd774507d5 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Pali=20Roh=C3=A1r?= <pali@kernel.org>
+Date: Wed, 20 May 2020 14:33:05 +0200
+Subject: [PATCH 6/6] iwinfo: add device id for Marvell 88W8997 SDIO wifi card
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Lets add device id for Marvell 88W8997 SDIO wifi card.
+
+Signed-off-by: Pali Rohár <pali@kernel.org>
+[added missing commit description]
+Signed-off-by: Petr Štetiar <ynezz@true.cz>
+(cherry picked from commit d577a9d38a3b544a70af5b6c7d91536f27e344f2)
+---
+ hardware.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/hardware.txt b/hardware.txt
+index d11d762..67ce81c 100644
+--- a/hardware.txt
++++ b/hardware.txt
+@@ -172,6 +172,7 @@
+ 0x11ab 0x2a55 0x11ab 0x0000    0      0  "Marvell"  "88W8864"
+ 0x02df 0x9135 0x0000 0x0000    0      0  "Marvell"  "88W8887"
+ 0x11ab 0x2b40 0x11ab 0x0000    0      0  "Marvell"  "88W8964"
++0x02df 0x9141 0x0000 0x0000    0      0  "Marvell"  "88W8997"
+ 0x14c3 0x7603 0x14c3 0x7603    0      0  "MediaTek" "MT7603E"
+ 0x14c3 0x7610 0x14c3 0x7610    0      0  "MediaTek" "MT7610E"
+ 0x14c3 0x7612 0x14c3 0x7612    0      0  "MediaTek" "MT7612E"
+-- 
+2.25.1
+
-- 
2.25.1

